service/status-report/main.go

```diff
@@ -138,10 +141,46 @@ func main() {
 	status := providerclient.NewStorageClientStatus().
 		SetPlatformVersion(pltVersion).
 		SetOperatorVersion(oprVersion)
-	if _, err = providerClient.ReportStatus(ctx, storageClient.Status.ConsumerID, status); err != nil {
+	statusResponse, err := providerClient.ReportStatus(ctx, storageClient.Status.ConsumerID, status)
+	if err != nil {
 		klog.Exitf("Failed to report status of storageClient %v: %v", storageClient.Status.ConsumerID, err)
 	}
 
+	versionFromResponse, err := getVersionFromChannel(statusResponse.DesiredClientOperatorChannel)
+	if err != nil {
+		klog.Exitf("Failed to map channel name from status response to version: %v", err)
+	}
```

Comment:
[nb-ohad](https://github.com/nb-ohad) [Mar 15, 2024](https://github.com/red-hat-storage/ocs-client-operator/pull/89#discussion_r1525440160)

Why are we doing this? why aren't we just saving the channel in the annotation?